### PR TITLE
[msal-angular][msal-browser] Typedocs and Instrumentation

### PR DESCRIPTION
**Pull request by jo-arroyo**
This PR: 
- Adds instrumentation to `msal-angular@2`
- Adds typedocs comments to `msal-angular@2`
- Adds instrumentation to `msal-browser`: `ClientApplication.ts` and `PublicClientApplication.ts`

### CODE REVIEW

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 1]
```diff
@@ -166,6 +170,7 @@ export abstract class ClientApplication {
 
         const responseHash = this.getRedirectResponseHash(hash || window.location.hash);
         if (!responseHash) {
+            this.logger.verbose("Hash returned not recognized or associated with redirect request, returning null");
```
Comment (tnorling) [inline lib/msal-browser/src/app/ClientApplication.ts:173]: Maybe redundant with the log message 2 lines down?

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 2]
```diff
@@ -214,6 +222,7 @@ export abstract class ClientApplication {
                 await BrowserUtils.navigateWindow(homepage, this.config.system.redirectNavigationTimeout, this.logger, true);
             } else {
                 // Navigate to page that initiated the redirect request
+                this.logger.verbose("Navigating to loginRequestUrl");
```
Comment (tnorling) [inline lib/msal-browser/src/app/ClientApplication.ts:225]:
```suggestion
                this.logger.verbose(`Navigating to loginRequestUrl: ${loginRequestUrl}`);
```

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 3]
```diff
@@ -552,6 +579,7 @@ export abstract class ClientApplication {
             const isInteractionRequiredError = e instanceof InteractionRequiredAuthError;
             const isInvalidGrantError = (e.errorCode === BrowserConstants.INVALID_GRANT_ERROR);
             if (isServerError && isInvalidGrantError && !isInteractionRequiredError) {
+                this.logger.verbose("Acquiring token by refresh token returned server and invalid grant error, attempting acquire token by iframe");
```
Comment (tnorling) [inline lib/msal-browser/src/app/ClientApplication.ts:582]:
```suggestion
                this.logger.verbose("Refresh token expired or invalid, attempting acquire token by iframe");
```

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 4]
```diff
@@ -647,8 +679,10 @@ export abstract class ClientApplication {
     getAccountByUsername(userName: string): AccountInfo|null {
         const allAccounts = this.getAllAccounts();
         if (!StringUtils.isEmpty(userName) && allAccounts && allAccounts.length) {
+            this.logger.verbose(`Returning signed-in accounts matching username: ${userName}`);
```
Comment (tnorling) [inline lib/msal-browser/src/app/ClientApplication.ts:682]: PII

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 5]
```diff
@@ -663,8 +697,10 @@ export abstract class ClientApplication {
     getAccountByHomeId(homeAccountId: string): AccountInfo|null {
         const allAccounts = this.getAllAccounts();
         if (!StringUtils.isEmpty(homeAccountId) && allAccounts && allAccounts.length) {
+            this.logger.verbose(`Returning signed-in accounts matching homeAccountId: ${homeAccountId}`);
```
Comment (tnorling) [inline lib/msal-browser/src/app/ClientApplication.ts:700]: PII

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 6]
```diff
@@ -679,8 +715,10 @@ export abstract class ClientApplication {
     getAccountByLocalId(localAccountId: string): AccountInfo | null {
         const allAccounts = this.getAllAccounts();
         if (!StringUtils.isEmpty(localAccountId) && allAccounts && allAccounts.length) {
+            this.logger.verbose(`Returning signed-in accounts matching localAccountId: ${localAccountId}`);
```
Comment (tnorling) [inline lib/msal-browser/src/app/ClientApplication.ts:718]: PII

File: lib/msal-browser/src/app/ClientApplication.ts (hunk 1 above)
Comment (jo-arroyo) [reply lib/msal-browser/src/app/ClientApplication.ts]: Yes! Must have missed it. Thanks for catching that. Removed

File: lib/msal-browser/src/app/ClientApplication.ts (hunk 4 above)
Comment (jo-arroyo) [reply lib/msal-browser/src/app/ClientApplication.ts]: Done

File: lib/msal-browser/src/app/ClientApplication.ts (hunk 5 above)
Comment (jo-arroyo) [reply lib/msal-browser/src/app/ClientApplication.ts]: Done

File: lib/msal-browser/src/app/ClientApplication.ts (hunk 6 above)
Comment (jo-arroyo) [reply lib/msal-browser/src/app/ClientApplication.ts]: Done

**Review by hectormmg — APPROVED** [review]

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 7]
```diff
@@ -647,8 +678,10 @@ export abstract class ClientApplication {
     getAccountByUsername(userName: string): AccountInfo|null {
         const allAccounts = this.getAllAccounts();
         if (!StringUtils.isEmpty(userName) && allAccounts && allAccounts.length) {
+            this.logger.verbosePii(`Returning signed-in accounts matching username: ${userName}`);
```
Comment (jasonnutter) [inline lib/msal-browser/src/app/ClientApplication.ts:681]: For every Pii log, can you also add a non-pii message next to it?

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 8]
```diff
@@ -690,14 +727,21 @@ export abstract class ClientApplication {
...
+            this.activeLocalAccountId = null;
+        }
     }
 
     /**
      * Gets the currently active account
      */
     getActiveAccount(): AccountInfo | null {
         if (!this.activeLocalAccountId) {
+            this.logger.verbose("No active account");
```
Comment (jasonnutter) [inline lib/msal-browser/src/app/ClientApplication.ts:744]:
```suggestion
            this.logger.verbose("getActiveAccount: No active account");
```

Comment (jasonnutter) [reply lib/msal-browser/src/app/ClientApplication.ts:744]: I think prefixing log messages like so can help make it easier to follow.

**Review by jasonnutter — APPROVED** [review]
A couple things, otherwise looks good!

File: lib/msal-browser/src/app/ClientApplication.ts (hunk 8 above)
Comment (jo-arroyo) [reply lib/msal-browser/src/app/ClientApplication.ts]: Makes sense, added

File: lib/msal-browser/src/app/ClientApplication.ts (hunk 7 above)
Comment (jo-arroyo) [reply lib/msal-browser/src/app/ClientApplication.ts:682]: Done

**Pull request merged**